File: rust/bubble-sort.rs
/// Sorts a list using bubble algorithm.
pub fn bubble_sort(t: Vec<usize>) -> Vec<usize> {
    let mut new_t = t;
    let mut sorted: bool = false;
    while sorted == false {
        sorted = true;
        for i in 0..new_t.len() - 1 {
            if new_t[i + 1] < new_t[i] {
                let temp = new_t[i + 1];
                new_t[i + 1] = new_t[i];
                new_t[i] = temp;
                sorted = false;
            }
        }
    }
    new_t
}

File: go/bubble-sort.go
// Sort a list using the Bubble sort algorithm.
func BubbleSort(t []float64) []float64 {
	sorted := false
	for sorted == false {
		sorted = true
		for i := 0; i < len(t)-1; i++ {
			if t[i] > t[i+1] {
				t[i+1], t[i] = t[i], t[i+1]
				sorted = false
			}
		}
	}
	return t
}